### PR TITLE
feature(C17): added solutions

### PR DESCRIPTION
**Pull request by rajatmodi62**
Solutions added for :
p07
p08
p09
p13
p16
p17
p18
p21
p22

### CODE REVIEW

**Comment by brycedrennan** [comment]
Do you want a code review or for us to cleanup the code? Either way is fine.

**Comment by rajatmodi62** [comment]
> Do you want a code review or for us to cleanup the code? Either way is fine.

Dear sir,
Many thanks for your reply. The code I have submitted works since I have already mentioned ctcis test cases there. 

Will be grateful if you could please cleanup the code. Meanwhile I will completely fill the gaps of this repo in terms of missing solutions on remaining hard and moderate problems
Thanks again

File: chapter_17/p08_circus_tower.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+#sorted list by height,weight respectively
+#first element is larger than next element
+#in simpler terms, trying to build the tower from bottom to up
+
+#a = [(65,100),(70,150),(56,90),(75,190),(60,95),(68,110)]
+a = [(65,100),(55,40),(75,90),(80,120)]
+#sort on second dimension in decreasing order
+a.sort(key= lambda x:-x[1])
```
Comment (brycedrennan) [inline chapter_17/p08_circus_tower.py:9]: all solutions should be in the form of a function

File: chapter_17/p21_volume_of_histogram.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        h_right[i]= max(h_right[i+1],a[i+1])
+    res = 0
+    #neglect extreme left and extreme right bar, they can never be filled
+    for i in range(1,n-1):
+        min_height = min(h_left[i],h_right[i])
+        if min_height-a[i]>0:
+            res+=min_height-a[i]
+    print(res)
+    return res
+find_volume(hist)
```
Comment (brycedrennan) [inline chapter_17/p21_volume_of_histogram.py:24]: any executing code should be wrapped in a `if __name__ == "__main__":`

**Comment by brycedrennan** [comment]
I've merged your solution to C17_P07 here: https://github.com/careercup/CtCI-6th-Edition-Python/commit/9c1bdef2ab1fd0883ddf91928b1035ee5e99ecd2
and did this cleanup: https://github.com/careercup/CtCI-6th-Edition-Python/commit/b36cf146ba89eda9e0396161a1c9cd81855a7399

Will review other solutions later.

**Comment by brycedrennan** [comment]
C17_P08 
merged: https://github.com/careercup/CtCI-6th-Edition-Python/commit/c60cbf4667b6c0e56c5eff9244d4c40e86b456d8
cleanup: https://github.com/careercup/CtCI-6th-Edition-Python/commit/0b6b20f79904d54704927d903111c5155d67dc13

**Comment by brycedrennan** [comment]
C17_P09
merged: https://github.com/careercup/CtCI-6th-Edition-Python/pull/194/commits/eae90c5c3195aa18dc289e9116e76fd5d9a026fc
cleanup:  https://github.com/careercup/CtCI-6th-Edition-Python/pull/194/commits/584890e2240ff76c466646324130078d02ccd203

**Review by brycedrennan — CHANGES_REQUESTED** [review]
the solution to problem 13 doesn't appear to be correct

**Comment by brycedrennan** [comment]
C17_P16
merged: https://github.com/careercup/CtCI-6th-Edition-Python/commit/ffca35842e27333287bcc28c4ee84ba7a683e2f6
cleanup: https://github.com/careercup/CtCI-6th-Edition-Python/commit/dde7f38bb3ac994563bac3268c39e27498230fd5

**Comment by brycedrennan** [comment]
C17_P17
merged: https://github.com/careercup/CtCI-6th-Edition-Python/commit/49525a58bdca2cf726419ca98531d5c84fa990b5
cleaned: https://github.com/careercup/CtCI-6th-Edition-Python/commit/36e62dd091ac1ebd25e70db505ebf702f47b4a54

**Comment by brycedrennan** [comment]
C17_P18
merged: https://github.com/careercup/CtCI-6th-Edition-Python/commit/c9f0cdbb4d5ace4e7870cd6406a00317d7f10dd5
cleaned: https://github.com/careercup/CtCI-6th-Edition-Python/commit/c9f0cdbb4d5ace4e7870cd6406a00317d7f10dd5

**Comment by brycedrennan** [comment]
C17_P22
merged: https://github.com/careercup/CtCI-6th-Edition-Python/commit/46f26c76e73ec8fdebde57dabc89cf8b2ba626ce
cleanup: https://github.com/careercup/CtCI-6th-Edition-Python/commit/0dc648e7c4757e17af0e32b4e637d95b86521567

**Comment by brycedrennan** [comment]
That takes care of this pull request. Please submit any future work in separate pull requests per solution.
